site/_components/stats.njk

```diff
@@ -4,14 +4,29 @@
   @param {object[]} items - List of objects containing the following attributes.
   @param {string} items[].stat - Statistic
   @param {string} items[].desc - Description of statistic
+  @param {object} items[].modifiers (optional) - Object containing the following attributes
+  @param {('semicircle' | 'circle' | 'triangle')} items[].modifiers.shape: The shape to be displayed behind the description term.
+  @param {('half' | 'full')} items[].modifiers.scale: The amount to scale the left value of the shape as the screen size shrinks.
+
+  @param {object} modifiers (optional) - An object with the following attributes.
+  @param {string} modifiers.heading - Section title.
+  @param {string} modifiers.subtitle - Text for a subtitle to be used in combination with a heading (modifiers.heading).
  #}-->
-{% macro stats(items) %}
-<figure class="stats">
+{% macro stats(items, modifiers={}) %}
+<figure class="stats {{'stats--header' if modifiers.heading is defined}}">
+  {% if modifiers.heading is defined %}
+  <header class="stats__header">
+    <h1 class="stats__title type--h1">{{modifiers.heading}}</h1>
+    {% if modifiers.subtitle %}
+    <p class="type--large stats__subtitle">{{modifiers.subtitle}}</p>
+    {% endif %}
+  </header>
+  {% endif %}
   <dl class="stats__list">
-    {% for item in items %}
+    {% for item in items %} {% if item.modifiers.shape is not defined %} {% set defaults = ['semicircle', 'circle', 'triangle'] %} {% set defaultShape = 'semicircle' %} {% set parentIndex = loop.index0 %} {% for shape in defaults %} {% if loop.index0 == parentIndex %} {% set defaultShape = shape %} {% endif %} {% endfor %} {% endif %}
     <div class="stats__item">
-      <dt class="stats__stat"><span>{{item.stat}}</span></dt>
-      <dd class="stats__desc type--h5">{{item.desc}}</dd>
+      <dt class="stats__stat {% if item.modifiers.shape is defined %} stats__stat--{{item.modifiers.shape}} {% else %} stats__stat--{{defaultShape}} {% endif %}{% if item.modifiers.scale is defined %} stats__stat--scale-{{item.modifiers.scale}} {% endif %}"><span>{{item.stat}}</span></dt>
```

Comment:
This looks like it can be simplified as follows (if I'm missing something, please LMK)
* Move the `defaults` set outside of the `for` loop (we don't need to redeclare it for each item
* Add a `set shape = ''` when the for loop starts
* if `item.modifiers.shape is not defined`, set `shape` to `defaults[loop.index0]`, otherwise set it to `item.modifiers.shape`
* Use `shape` directly instead of querying for it on line 28